### ISSUE TITLE
`query` with stdlib functions applications returns some very strange errors

### ISSUE THREAD

**Issue by regnat** [tweag/nickel#425]
**Describe the bug**

Most applications of a function from the stdlib raise a strange error when merged with something or ran through `nickel query`. For example:

```console
$ nickel query <<< 'builtins.seq 1 1'
error: Not a function
    ┌─ <stdlib/builtins.ncl>:127:7
    │
127 │     = fun x y => %seq% x y,
    │       ^^^^^^^^^^^^^^^^^^^^ this term is applied, but it is not a function
    │
    ┌─ <stdin>:1:1
    │
  1 │ builtins.seq 1 1
    │ -------------- applied here
```

(Applications of `lists.map`, `records.fieldsOf` and others return a similar `Not a function` error on their respective definition site)

**Environment**
 - OS name + version: NixOS 21.05
 - Version of the code: 62c5d83bb5a4204895fc5fa8eefe95cf97265364

**Comment by regnat** [comment]
(just to clarify the “merge with something bit”, `nickel export <<< '{ x = builtins.seq 1 1 }' ` works just fine, but `nickel export <<< '{ x = builtins.seq 1 1 } & {x | Num}'` raises the error

**Comment by yannham** [comment]
Should be fixed by #433.
